lib/cloud/file-storage/awsStorage.js

```diff
@@ -0,0 +1,91 @@
+var q = require('q');
+var s3 = require('s3');
+var path = require('path');
+
+var config;
+var awsClient;
+
+function init(storageConfiguration) {
+  config = storageConfiguration;
+  validateConfig();
+  awsClient = s3.createClient(config);
+}
+
+function writeFile(namespace, fileName, fileLocation) {
+  var file;
+  if (namespace) {
+    file = path.join(namespace, fileName);
+  } else {
+    file = fileName;
+  }
+  
+  var params = {
+    localFile: fileLocation,
+    ACL: 'authenticated-read',
+    s3Params: {
+      Bucket: config.bucket,
+      Key: file
+    }
+  };
+  
+  var deferred = q.defer();
+  var uploader = awsClient.uploadFile(params);
+  uploader.on('error', function(err) {
+    console.log(err);
+    deferred.reject(err.stack);
+  });
+  uploader.on('progress', function() {
```

Comment:
Are we doing anything with this `progress` event? If not we can probably remove the listener.